notebooks/learning-to-rank/01-learning-to-rank.ipynb

```diff
@@ -0,0 +1,1513 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "TKxL_NmZmOpF"
+   },
+   "source": [
+    "# How to train and deploy Learning To Rank\n",
+    "\n",
+    "TODO: udpate the link to elastic/elasticsearch-labs instead of my fork before merging.\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/afoucret/elasticsearch-labs/blob/ltr-notebook/notebooks/learning-to-rank/01-learning-to-rank.ipynb)\n",
+    "\n",
+    "In this notebook we will see an example on how to train a Learning To Rank model using [XGBoost](https://xgboost.ai/) and how to deploy it to be used as a rescorer in Elasticsearch.\n",
+    "\n",
+    "\n",
+    "**Notes about the Learning To Rank feature:**\n",
+    "- The Learning To Rank feature is available for Elastic Stack versions 8.12.0 and newer and requires a Platinum subscription or higher.\n",
+    "- The Learning To rank is experimental and may be changed or removed completely in future releases. Elastic will make a best effort to fix any issues, but experimental features are not supported to the same level as generally available (GA) features.\n",
+    " \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Jq6mztWOmOpH"
+   },
+   "source": [
+    "## Install required packages\n",
+    "\n",
+    "First we will be installing packages required for our example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 99,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "0nCl2nhamOpH",
+    "outputId": "1e7380e7-4944-430a-db5f-180f1e299615"
+   },
+   "outputs": [],
+   "source": [
+    "# TODO: when eland 8.12.1 is released, we can avoid installing from github main:\n",
+    "!pip install -qU git+https://github.com/elastic/eland@main\n",
+    "!pip install -qU elasticsearch \"eland[scikit-learn]\" xgboost tqdm\n",
+    "\n",
+    "from tqdm import tqdm\n",
+    "\n",
+    "# Setup the progress bar so we can use progress_apply in the notebook.\n",
+    "tqdm.pandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "yks44hf0mOpI"
+   },
+   "source": [
+    "## Configure your Elasticsearch deployment\n",
+    "\n",
+    "For this example, we will be using an [Elastic Cloud](https://www.elastic.co/guide/en/cloud/current/ec-getting-started.html) deployment (available with a [free trial](https://cloud.elastic.co/registration?utm_source=github&utm_content=elasticsearch-labs-notebook))."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 100,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 71
+    },
+    "id": "IpnP7JUHmOpI",
+    "outputId": "eb52c692-a773-4863-f930-fdedb5c6e0eb"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Successfully connected to cluster bd63f706e18b476aacb5cb0aaeb5f0bd (version 8.12.0)'"
```

Comment:
not sure if it makes sense to display your actual cluster ID here. probably worth removing it from the string or clearing the output

Reply:
This is something that allow me to know in one sight where the notebook is running. I would prefer to keep it.

Reply:
That's fine, it is just recommended to clear the output of the cell then so that [nbtest](https://github.com/elastic/elasticsearch-labs/blob/444e05a42fad8d36939f5c9b133088662e5edc2b/CONTRIBUTING.md#automated-testing) does not complain about it. Which reminds me that it would be great to add this notebook to the list of notebooks that should be tested:
https://github.com/elastic/elasticsearch-labs/blob/444e05a42fad8d36939f5c9b133088662e5edc2b/notebooks/search/Makefile#L2-L9